Multisig/UI/WalletConnectV2/WalletConnectManager.swift

```diff
@@ -73,140 +73,184 @@ class WalletConnectManager {
                 let proposalNamespace = $0.value
-                let chains = proposalNamespace.chains
-
-                let accounts = Set(chains.compactMap {
+                guard let chains = proposalNamespace.chains else { return }
+                
+                let selectedSafeChain = chains.filter { chain in
+                    chain.namespace == "eip155" && chain.reference == safe.chain?.id
+                }
+                
+                let accounts = Set(selectedSafeChain.compactMap {
                     Account($0.absoluteString + ":\(safe.addressValue)")
                 })
-
+                
                 let sessionNamespace = SessionNamespace(accounts: accounts,
                                                         methods: proposalNamespace.methods,
                                                         events: proposalNamespace.events)
                 sessionNamespaces[caip2Namespace] = sessionNamespace
             }
             do {
-                try await Sign.instance.approve(proposalId: proposal.id, namespaces: sessionNamespaces)
+                try await Web3Wallet.instance.approve(proposalId: proposal.id, namespaces: sessionNamespaces)
             } catch {
                 print("DAPP: Approve Session error: \(error)")
+                Task { @MainActor in
+                    App.shared.snackbar.show(message: "DAPP: Approve Session error: \(error)")
+                }
             }
         }
     }
-
+    
     /// By default, session lifetime is set for 7 days and after that time user's session will expire.
     /// This method will extend the session for 7 days
     func extend(session: Session) async {
         do {
             try await Web3Wallet.instance.extend(topic: session.topic)
         } catch {
+            Task { @MainActor in
+                App.shared.snackbar.show(message: error.localizedDescription)
+            }
             print("DAPP: extending Session error: \(error)")
         }
     }
-
+    
     func disconnect(session: Session) {
         Task {
             do {
                 NotificationCenter.default.post(name: .wcDidDisconnectSafeServer, object: self)
-                try await Sign.instance.disconnect(topic: session.topic)
+                try await Web3Wallet.instance.disconnect(topic: session.topic)
             } catch {
-                print("DAPP: disconnectting Session error: \(error)")
+                print("DAPP: disconnecting Session error: \(error)")
+                Task { @MainActor in
+                    App.shared.snackbar.show(message: "DAPP: disconnecting Session error: \(error)")
+                }
             }
+            disconnectUnusedPairings()
         }
     }
-
+    
     func deleteStoredSession(topic: String) {
         precondition(Thread.isMainThread)
         Safe.removeSession(topic: topic)
+        disconnectUnusedPairings()
     }
-
+    
+    private func disconnectUnusedPairings() {
```

Comment:
Sorry but I don't think this should be the way to go for this, we need to discover why exactly this happens, please check the code of WalletConnect library, it checks for the validity of session and pairing each time on fetching,

Reply:
What do you mean by: "we need to discover why exactly this happens"?

When the session is connected remotely (from the Dapp side) we only get the topic of the session after the session is already disconnected and deleted. So we do not have the pairingTopic. We agreed that we want to disconnect the pairing when a session is disconnected. I do not see any other way to achieve this than to iterate over the sessions to find the unused pairings.